src/librustc_mir/interpret/memory.rs

```diff
@@ -648,59 +648,14 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
         &self,
         allocs_seen: &mut FxHashSet<AllocId>,
         allocs_to_print: &mut VecDeque<AllocId>,
-        mut msg: String,
```

Comment:
cc @RalfJung 

I ripped out the allocation formatter and put it into `pretty.rs` and am using it from here. There was a single formatting change (check out the MIR dumps at the end of this commit to see how it will look in miri, too)